src/main/java/network/brightspots/rcv/Tabulator.java

```diff
@@ -72,17 +72,24 @@ class Tabulator {
   // tracks residual surplus from multi-seat contest vote transfers
   private final Map<Integer, BigDecimal> roundToResidualSurplus = new HashMap<>();
   // precincts which may appear in the cast vote records
-  private final Set<String> precinctNames;
+  private final Set<String> precinctNames = new HashSet<>();
   // tracks the current round (and when tabulation is completed, the total number of rounds)
   private int currentRound = 0;
   // tracks required winning threshold
   private BigDecimal winningThreshold;
 
-  Tabulator(List<CastVoteRecord> castVoteRecords, ContestConfig config, Set<String> precinctNames) {
+  Tabulator(List<CastVoteRecord> castVoteRecords, ContestConfig config) {
     this.castVoteRecords = castVoteRecords;
     this.candidateNames = config.getCandidateNames();
     this.config = config;
-    this.precinctNames = precinctNames;
+
+    for (CastVoteRecord cvr : castVoteRecords) {
+      String precinctName = cvr.getPrecinct();
+      if (precinctName != null) {
+        precinctNames.add(precinctName);
+      }
+    }
```

Comment:
Ok, I understand that this is now where it's being calculated, but I'm absolutely struggling to find where it _used_ to be calculated...

It seems like when we're moving the calculation to here, we should also be getting rid of wherever it was calculated before, and I can't see anywhere in this PR where that's being done.

Reply:
Scratch that, I now see the single line where it's added in `StreamingCvrReader`. And I think I understand the motivation for this PR (cleaning up a bunch of stuff in the signatures that wasn't used in most cases).

However, isn't iterating through every single CVR a pretty intense thing, say, if you have millions of votes? It seems like it's much more efficient to do it where it was done previously (plus `if (precinctColumnIndex != null) {` was protecting it further).

It seems to me like there should be a way to engineer this so we get the best of both worlds.

Reply:
This is quite fast since it does no computation. Running through a million numbers is nearly instant once that data is in memory.

I ran a quick test just in case you're concerned, surrounding this block with START/END:
```
2023-06-06 20:26:52 EDT INFO: Reading Dominion cast vote records from: <>
2023-06-06 20:26:59 EDT INFO: Parsed 50025 records from 700 files
2023-06-06 20:27:06 EDT INFO: Parsed 100044 records from 1395 files
2023-06-06 20:27:13 EDT INFO: Parsed 150099 records from 2211 files
2023-06-06 20:27:20 EDT INFO: Parsed 200187 records from 3012 files
2023-06-06 20:27:29 EDT INFO: Parsed 250268 records from 3914 files
2023-06-06 20:27:35 EDT INFO: Parsed 300338 records from 4586 files
2023-06-06 20:27:41 EDT INFO: Parsed 350374 records from 5168 files
2023-06-06 20:27:47 EDT INFO: Parsed 400442 records from 5754 files
2023-06-06 20:27:53 EDT INFO: Parsed 450464 records from 6493 files
2023-06-06 20:28:00 EDT INFO: Parsed 500555 records from 7198 files
2023-06-06 20:28:06 EDT INFO: Parsed 550611 records from 7867 files
2023-06-06 20:28:12 EDT INFO: Parsed 600651 records from 8545 files
2023-06-06 20:28:19 EDT INFO: Parsed 650709 records from 9289 files
2023-06-06 20:28:25 EDT INFO: Parsed 700787 records from 9894 files
2023-06-06 20:28:31 EDT INFO: Parsed 750833 records from 10553 files
2023-06-06 20:28:37 EDT INFO: Parsed 800858 records from 11201 files
2023-06-06 20:28:42 EDT INFO: Parsed 850863 records from 11818 files
2023-06-06 20:28:48 EDT INFO: Parsed 900897 records from 12479 files
2023-06-06 20:28:55 EDT INFO: Parsed 950956 records from 13116 files
2023-06-06 20:29:01 EDT INFO: Parsed 1001049 records from 13766 files
2023-06-06 20:29:08 EDT INFO: Parsed 1051081 records from 14449 files
2023-06-06 20:29:14 EDT INFO: Parsed 1101089 records from 15057 files
2023-06-06 20:29:20 EDT INFO: Parsed 1151115 records from 15684 files
2023-06-06 20:29:26 EDT INFO: Parsed 1201207 records from 16298 files
2023-06-06 20:29:31 EDT INFO: Parsed 1251212 records from 16921 files
2023-06-06 20:29:38 EDT INFO: Parsed 1301226 records from 17567 files
2023-06-06 20:29:47 EDT INFO: Parsed 1351244 records from 18765 files
2023-06-06 20:30:01 EDT INFO: Parsed 1401391 records from 20504 files
2023-06-06 20:30:03 EDT INFO: Parsed 1451707 records from 20695 files
2023-06-06 20:30:06 EDT INFO: Parsed 1501820 records from 20877 files
2023-06-06 20:30:16 EDT INFO: Writing cast vote records in generic format to file: 
2023-06-06 20:30:16 EDT INFO: Successfully wrote: 
2023-06-06 20:30:16 EDT INFO: Parsed 308340 cast vote records successfully.
2023-06-06 20:30:16 EDT INFO: START
2023-06-06 20:30:16 EDT INFO: END
```

In other words, it takes over 3 minutes to read the cast vote records for this election with 1.5m records, and under a second to do this calculation. I'm not sure it's worth optimizing.

Reply:
Nice check! Confirmed on my end as well with `2017_minneapolis_mayor` and it was similarly fast.